### ISSUE TITLE
create_report() crashes on a dataframe with a constant numeric column

### ISSUE THREAD

**Issue by dhuntley1023** [sfu-db/dataprep#464]
create_report() crashes when the dataframe contains a column of all constant numeric values.  Interestingly, I didn't see the failure when I used plot(), plot_correlation(), or plot_missing(), so maybe they have logic to adapt to that condition.

Below is a simple repro case.

Repro:
```
dp.create_report(pd.DataFrame({'A': [1,2,3], 'B': [1,1,1]}))
```

Error trace:
```
---------------------------------------------------------------------------
LinAlgError                               Traceback (most recent call last)
<ipython-input-65-6594a8550d86> in <module>
----> 1 dp.create_report(pd.DataFrame({'A': [1,2,3], 'B': [1,1,1]}))

c:\Miniconda\envs\python3\lib\site-packages\dataprep\eda\create_report\__init__.py in create_report(df, title, mode, progress)
     52         "resources": INLINE.render(),
     53         "title": title,
---> 54         "components": format_report(df, mode, progress),
     55     }
     56     template_base = ENV_LOADER.get_template("base.html")

c:\Miniconda\envs\python3\lib\site-packages\dataprep\eda\create_report\formatter.py in format_report(df, mode, progress)
     61         df = string_dtype_to_object(df)
     62         if mode == "basic":
---> 63             comps = format_basic(df)
     64         # elif mode == "full":
     65         #     comps = format_full(df)

c:\Miniconda\envs\python3\lib\site-packages\dataprep\eda\create_report\formatter.py in format_basic(df)
     97             category=RuntimeWarning,
     98         )
---> 99         (data,) = dask.compute(data)
    100 
    101     # results dictionary

c:\Miniconda\envs\python3\lib\site-packages\dask\base.py in compute(*args, **kwargs)
    450         postcomputes.append(x.__dask_postcompute__())
    451 
--> 452     results = schedule(dsk, keys, **kwargs)
    453     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    454 

c:\Miniconda\envs\python3\lib\site-packages\dask\threaded.py in get(dsk, result, cache, num_workers, pool, **kwargs)
     82         get_id=_thread_get_id,
     83         pack_exception=pack_exception,
---> 84         **kwargs
     85     )
     86 

c:\Miniconda\envs\python3\lib\site-packages\dask\local.py in get_async(apply_async, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, **kwargs)
    484                         _execute_task(task, data)  # Re-execute locally
    485                     else:
--> 486                         raise_exception(exc, tb)
    487                 res, worker_id = loads(res_info)
    488                 state["cache"][key] = res

c:\Miniconda\envs\python3\lib\site-packages\dask\local.py in reraise(exc, tb)
    314     if exc.__traceback__ is not tb:
    315         raise exc.with_traceback(tb)
--> 316     raise exc
    317 
    318 

c:\Miniconda\envs\python3\lib\site-packages\dask\local.py in execute_task(key, task_info, dumps, loads, get_id, pack_exception)
    220     try:
    221         task, data = loads(task_info)
--> 222         result = _execute_task(task, data)
    223         id = get_id()
    224         result = dumps((result, id))

c:\Miniconda\envs\python3\lib\site-packages\dask\core.py in _execute_task(arg, cache, dsk)
    119         # temporaries by their reference count and can execute certain
    120         # operations in-place.
--> 121         return func(*(_execute_task(a, cache) for a in args))
    122     elif not ishashable(arg):
    123         return arg

c:\Miniconda\envs\python3\lib\site-packages\dataprep\eda\distribution\compute\common.py in gaussian_kde(arr)
    230 def gaussian_kde(arr: np.ndarray) -> Tuple[float, float]:
    231     """Delayed version of scipy gaussian_kde."""
--> 232     return cast(Tuple[np.ndarray, np.ndarray], gaussian_kde_(arr))
    233 
    234 

c:\Miniconda\envs\python3\lib\site-packages\scipy\stats\kde.py in __init__(self, dataset, bw_method, weights)
    204             self._neff = 1/sum(self._weights**2)
    205 
--> 206         self.set_bandwidth(bw_method=bw_method)
    207 
    208     def evaluate(self, points):

c:\Miniconda\envs\python3\lib\site-packages\scipy\stats\kde.py in set_bandwidth(self, bw_method)
    554             raise ValueError(msg)
    555 
--> 556         self._compute_covariance()
    557 
    558     def _compute_covariance(self):

c:\Miniconda\envs\python3\lib\site-packages\scipy\stats\kde.py in _compute_covariance(self)
    566                                                bias=False,
    567                                                aweights=self.weights))
--> 568             self._data_inv_cov = linalg.inv(self._data_covariance)
    569 
    570         self.covariance = self._data_covariance * self.factor**2

c:\Miniconda\envs\python3\lib\site-packages\scipy\linalg\basic.py in inv(a, overwrite_a, check_finite)
    975         inv_a, info = getri(lu, piv, lwork=lwork, overwrite_lu=1)
    976     if info > 0:
--> 977         raise LinAlgError("singular matrix")
    978     if info < 0:
    979         raise ValueError('illegal value in %d-th argument of internal '

LinAlgError: singular matrix

```

**Comment by brandonlockhart** [comment]
Thanks @dhuntley1023 for creating this issue! This is a known problem (related to #230): column B is treated as numerical, and then we try to compute a KDE which throws an error since the column is singular. This should be fixed in the near future when we improve our type detection.

**Comment by dovahcrow** [comment]
closed in favor of #230

**Comment by jspencer1289** [comment]
Hi guys, Im getting a similar issue - is there anyway to get around it or is it just waiting for the fix? Thanks! I can post error if need be

**Comment by jnwang** [comment]
Hi @jspencer1289, can I ask which dataprep version you are using? This bug should have been fixed in dataprep 0.3. You can install this latest version via `pip install -U dataprep`.

**Comment by jspencer1289** [comment]
Thanks a lot @jnwang. I just installed it within anaconda yesterday so I didn't check if it was the latest or not. I will update and let you know. Thanks!

**Comment by jspencer1289** [comment]
@jnwang Will `pip install -U dataprep` work with Anaconda do you know? Thanks

**Comment by jinglinpeng** [comment]
Hi @jspencer1289 , Anaconda is not the latest version and @dovahcrow  is working on the update. Have you tried the `pip`?

**Comment by jspencer1289** [comment]
I got the following error when running the `pip` @jinglinpeng 

```
ERROR: Could not install packages due to an EnvironmentError: [WinError 5] Access is denied: 'c:\\users\\latitude user\\anaconda3\\lib\\site-packages\\~andas\\_libs\\algos.cp37-win_amd64.pyd'
Consider using the `--user` option or check the permissions.
```

**Comment by dovahcrow** [comment]
> ERROR: Could not install packages due to an EnvironmentError: [WinError 5] Access is denied: 'c:\\users\\latitude user\\anaconda3\\lib\\site-packages\\~andas\\_libs\\algos.cp37-win_amd64.pyd'
> Consider using the `--user` option or check the permissions.

How about adding the `--user` option? i.e. `pip install -U --user dataprep`

**Comment by mparoca** [comment]
> Hi @jspencer1289, can I ask which dataprep version you are using? This bug should have been fixed in dataprep 0.3. You can install this latest version via `pip install -U dataprep`.

Hello, I'm using dataprep version 0.3.0 and I'm getting the same error as in the original post.  

Error trace:

`---------------------------------------------------------------------------
LinAlgError                               Traceback (most recent call last)
<ipython-input-23-af366d2f7f35> in <module>
----> 1 report = create_report(itrc_sample)

~\anaconda3\envs\transparencia3\lib\site-packages\dataprep\eda\create_report\__init__.py in create_report(df, config, display, title, mode, progress)
     66         "resources": INLINE.render(),
     67         "title": title,
---> 68         "components": format_report(df, cfg, mode, progress),
     69     }
     70     template_base = ENV_LOADER.get_template("base.html")

~\anaconda3\envs\transparencia3\lib\site-packages\dataprep\eda\create_report\formatter.py in format_report(df, cfg, mode, progress)
     71         if mode == "basic":
     72             edaframe = EDAFrame(df)
---> 73             comps = format_basic(edaframe, cfg)
     74         # elif mode == "full":
     75         #     comps = format_full(df)

~\anaconda3\envs\transparencia3\lib\site-packages\dataprep\eda\create_report\formatter.py in format_basic(df, cfg)
    117             category=RuntimeWarning,
    118         )
--> 119         (data,) = dask.compute(data)
    120     # results dictionary
    121     res: Dict[str, Any] = {}

~\anaconda3\envs\transparencia3\lib\site-packages\dask\base.py in compute(*args, **kwargs)
    450         postcomputes.append(x.__dask_postcompute__())
    451 
--> 452     results = schedule(dsk, keys, **kwargs)
    453     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    454 

~\anaconda3\envs\transparencia3\lib\site-packages\dask\threaded.py in get(dsk, result, cache, num_workers, pool, **kwargs)
     82         get_id=_thread_get_id,
     83         pack_exception=pack_exception,
---> 84         **kwargs
     85     )
     86 

~\anaconda3\envs\transparencia3\lib\site-packages\dask\local.py in get_async(apply_async, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, **kwargs)
    484                         _execute_task(task, data)  # Re-execute locally
    485                     else:
--> 486                         raise_exception(exc, tb)
    487                 res, worker_id = loads(res_info)
    488                 state["cache"][key] = res

~\anaconda3\envs\transparencia3\lib\site-packages\dask\local.py in reraise(exc, tb)
    314     if exc.__traceback__ is not tb:
    315         raise exc.with_traceback(tb)
--> 316     raise exc
    317 
    318 

~\anaconda3\envs\transparencia3\lib\site-packages\dask\local.py in execute_task(key, task_info, dumps, loads, get_id, pack_exception)
    220     try:
    221         task, data = loads(task_info)
--> 222         result = _execute_task(task, data)
    223         id = get_id()
    224         result = dumps((result, id))

~\anaconda3\envs\transparencia3\lib\site-packages\dask\core.py in _execute_task(arg, cache, dsk)
    119         # temporaries by their reference count and can execute certain
    120         # operations in-place.
--> 121         return func(*(_execute_task(a, cache) for a in args))
    122     elif not ishashable(arg):
    123         return arg

~\anaconda3\envs\transparencia3\lib\site-packages\dataprep\eda\utils.py in gaussian_kde(arr)
    381 def gaussian_kde(arr: np.ndarray) -> Tuple[float, float]:
    382     """Delayed version of scipy gaussian_kde."""
--> 383     return cast(Tuple[np.ndarray, np.ndarray], gaussian_kde_(arr))
    384 
    385 

~\anaconda3\envs\transparencia3\lib\site-packages\scipy\stats\kde.py in __init__(self, dataset, bw_method, weights)
    204             self._neff = 1/sum(self._weights**2)
    205 
--> 206         self.set_bandwidth(bw_method=bw_method)
    207 
    208     def evaluate(self, points):

~\anaconda3\envs\transparencia3\lib\site-packages\scipy\stats\kde.py in set_bandwidth(self, bw_method)
    554             raise ValueError(msg)
    555 
--> 556         self._compute_covariance()
    557 
    558     def _compute_covariance(self):

~\anaconda3\envs\transparencia3\lib\site-packages\scipy\stats\kde.py in _compute_covariance(self)
    566                                                bias=False,
    567                                                aweights=self.weights))
--> 568             self._data_inv_cov = linalg.inv(self._data_covariance)
    569 
    570         self.covariance = self._data_covariance * self.factor**2

~\anaconda3\envs\transparencia3\lib\site-packages\scipy\linalg\basic.py in inv(a, overwrite_a, check_finite)
    961         inv_a, info = getri(lu, piv, lwork=lwork, overwrite_lu=1)
    962     if info > 0:
--> 963         raise LinAlgError("singular matrix")
    964     if info < 0:
    965         raise ValueError('illegal value in %d-th argument of internal '

LinAlgError: singular matrix
`

My dataset is really large (149 variables), and I just wish I could see which variable was causing the issue to fix or remove it. 

Would this be solved if I specified the data types when using the create_report function? If so, do you know where I could find an example of the data types being specified within the create_report function? It wasn't clear from the documentation.

**Comment by jspencer1289** [comment]
> > ERROR: Could not install packages due to an EnvironmentError: [WinError 5] Access is denied: 'c:\users\latitude user\anaconda3\lib\site-packages\~andas\_libs\algos.cp37-win_amd64.pyd'
> > Consider using the `--user` option or check the permissions.
> 
> How about adding the `--user` option? i.e. `pip install -U --user dataprep`

All working now for me, thanks!!
